openmc/model/surface_composite.py

```diff
@@ -52,6 +53,168 @@ def __neg__(self):
         """Return the negative half-space of the composite surface."""
 
 
+class CylinderSector(CompositeSurface):
+    """Infinite cylindrical sector composite surface.
+
+    A cylinder sector is composed of two cylindrical and two planar surfaces.
+    The cylindrical surfaces are concentric, and the planar surfaces intersect
+    the central axis of the cylindrical surfaces.
+
+    This class acts as a proper surface, meaning that unary `+` and `-`
+    operators applied to it will produce a half-space. The negative
+    side is defined to be the region inside of the cylinder sector.
+
+    Parameters
+    ----------
+    r1 : float
+        Inner radius of sector. Must be less than r2.
+    r2 : float
+        Outer radius of sector. Must be greater than r1.
+    theta1 : float
+        Clockwise-most bound of sector in degrees. Assumed to be in the
+        counterclockwise direction with respect to the first basis axis
+        (+y, +z, or +x). Must be less than :attr:`theta2`.
+    theta2 : float
+        Counterclockwise-most bound of sector in degrees. Assumed to be in the
+        counterclockwise direction with respect to the first basis axis
+        (+y, +z, or +x). Must be greater than :attr:`theta1`.
+    center : iterable of float
+       Coordinate for central axes of cylinders in the (y, z), (z, x), or (x, y)
+       basis. Defaults to (0,0).
+
+    axis : {'x', 'y', 'z'}
+        Central axis of the cylinders defining the inner and outer surfaces of
+        the sector. Defaults to 'z'.
+    kwargs : dict
+            Keyword arguments passed to the :class:`Cylinder` and :class:`Plane` constructors.
+
+    Attributes
+    ----------
+    outer_cyl : openmc.ZCylinder, openmc.YCylinder, or openmc.XCylinder
+        Outer cylinder surface.
+    inner_cyl : openmc.ZCylinder, openmc.YCylinder, or openmc.XCylinder
+        Inner cylinder surface.
+    plane1 : openmc.Plane
+        Plane at angle :math:`\\theta_1` relative to the first basis axis.
+    plane2 : openmc.Plane
+        Plane at angle :math:`\\theta_2` relative to the first basis axis.
+
+    """
+
+    _surface_names = ('outer_cyl', 'inner_cyl', 'plane1', 'plane2')
+
+    def __init__(self,
+                 r1,
+                 r2,
+                 theta1,
+                 theta2,
+                 center=(0.,0.),
+                 axis='z',
+                 **kwargs):
+
+        if r2 <= r1 * sqrt(2):
+            raise ValueError(f'r2 must be greater than r1.')
+
+        if theta2 <= theta1:
+            raise ValueError(f'theta2 must be greater than theta1.')
+
+        self._theta1 = theta1
+        self._theta2 = theta2
+
```

Comment:
It doesn't look like these attributes are actually used other than in a test. For the purposes of testing the alternate constructor, I would create two CylinderSector surfaces, one with `__init__` and one with `from_theta_alpha` and then check that `surf.plane1.coefficients` and `surf.plane2.coefficients` are the same for both.
```suggestion
```